crates/cairo-lint-core/tests/ifs/collapsible_if.rs

```diff
@@ -148,6 +148,74 @@ fn main() {
 }
 "#;
 
+const IF_LET_TO_IGNORE_WITH_ASSERT: &str = r#"
+fn main() {
+    let x = Option::Some(true);
+    let y = Option::Some(true);
+
+    if let Option::Some(_z) = x {
+        assert!(x == y);
+    }
+}
+"#;
+
+const COLLAPSIBLE_IFS_INSIDE_IF_LET: &str = r#"
+fn main() {
+    let x = Option::Some(true);
+
+    let a = true;
+    let b = true;
+    let c = false;
+
+    if let Option::Some(_y) = x {
+         if a || b {
+            if b && c {
+                println!("Hello");
+            }
+        }
+    }
+}
+"#;
+
+const SIMPLE_IF_INSIDE_IF_LET: &str = r#"
+fn main() {
+    let x = Option::Some(true);
+
+    let a = true;
+    let b = true;
+
+    if let Option::Some(_y) = x {
+        if a || b {
+            println!("Hello");
+        }
+    }
+}
+"#;
+
+const COLLAPSIBLE_IF_LETS: &str = r#"
+fn main() {
+    let x = Some(Some(42));
+
+    if let Some(inner) = x {
+        if let Some(value) = inner {
+            println!("The value is: {}", value);
+        }
+    }
+}
+"#;
+
+const IF_WITH_ASSERT: &str = r#"
+fn main() {
+    let x = Some(42);
+    let y = Some(2);
+    let z = Some(10)
+
+    if x == y {
+        assert!(z == Some(42));
+    }
+}
+"#;
+
```

Comment:
Please add a test with `if let` nested within the `if`. Just to cover it so we can always be sure about that case